sklearn/preprocessing/_discretization.py

```diff
@@ -63,6 +66,26 @@ class KBinsDiscretizer(TransformerMixin, BaseEstimator):
 
         .. versionadded:: 0.24
 
+    subsample : int or None (default='warn')
+        Maximum number of samples, used to fit the model, for computational
+        efficiency. Used when `strategy="quantile"`.
+        `subsample=None` means that all the training samples are used when
+        computing the quantiles that determine the binning thresholds.
+        Since quantile computation relies on sorting each column of `X` and
+        that sorting has an `n log(n)` time complexity,
+        it is recommended to use subsampling on datasets with a
+        very large number of samples.
+
+        .. versionadded:: 1.1
```

Comment:
We might want to mention that it will be a changed in 2 versions

```
.. deprecated:: 1.1
   In version 1.3 and onwards, `subsample=2e5` will be the default.
```